activemodel/lib/active_model/type/decimal_without_scale.rb

```diff
@@ -6,6 +6,10 @@ class DecimalWithoutScale < BigInteger # :nodoc:
       def type
         :decimal
       end
+
+      def type_cast_for_schema(value)
+        value.to_s.inspect
```

Comment:
https://github.com/rails/rails/pull/25296#discussion_r66135824
> This class shouldn't need to implement this method, as it does not return `BigDecimal` objects.

`type_cast_for_schema` is only called by `ColumnDumper#schema_default`:

https://github.com/rails/rails/blob/v5.0.0.rc1/activerecord/lib/active_record/connection_adapters/abstract/schema_dumper.rb#L86-L94

```ruby
      def schema_default(column)
        type = lookup_cast_type_from_column(column)
        default = type.deserialize(column.default)
        if default.nil?
          schema_expression(column)
        else
          type.type_cast_for_schema(default)
        end
      end
```

`DecimalWithoutScale#type_cast_for_schema` is passed an integer value by `type.deserialize(column.default)` because the class inherits `BigInteger`.

Should we dump a default as a raw integer when `decimal` without scale?

Before:

```ruby
  create_table "defaults", force: :cascade do |t|
    t.string   "string_with_default",   default: "Hello!"
    t.date     "date_with_default",     default: '2014-06-05'
    t.datetime "datetime_with_default", default: '2014-06-05 07:17:04'
    t.time     "time_with_default",     default: '2000-01-01 07:17:04'
    t.decimal  "decimal_with_default",  default: 1234567890
  end
```

After:

```ruby
  create_table "defaults", force: :cascade do |t|
    t.string   "string_with_default",   default: "Hello!"
    t.date     "date_with_default",     default: "2014-06-05"
    t.datetime "datetime_with_default", default: "2014-06-05 07:17:04"
    t.time     "time_with_default",     default: "2000-01-01 07:17:04"
    t.decimal  "decimal_with_default",  default: "1234567890"
  end
```